### PR TITLE
Goodreads: Search by Author/Title if ASIN look up fails

### LINKED ISSUE #24: Goodreads look up failure - attempt requery
When no results are returned for ASIN, attempt a requery with author and title.

You can assign this to me.

### PR DESCRIPTION
**Pull request by SenorSmartyPants**
This fixes #24

Move all logic to audible.py
Only API access in api.py
logs debug messages
compares author/title results to verify match

### CODE REVIEW

File: audible.py [hunk 1]
```diff
@@ -378,6 +379,59 @@ def get_album_info(self, asin):
...
+                return work
+
+    def goodreads_get_total_result(self, response: Dict) -> int:
+        return int(response.findtext("./search/total-results"))
+
+    def parse_original_date(self, work: Dict) -> Dict:
+        original_date = {}
+        if work is not None:
+            original_date["year"] = int(work.findtext("original_publication_year"))
+            original_date["month"] = int(work.findtext("original_publication_month"))
```
[the remark this answers is not in the document]
Comment (SenorSmartyPants) [reply audible.py:431]: Ok. I'll take a look at doing that.

[the remark this answers is not in the document]
Comment (SenorSmartyPants) [reply audible.py]: Committed

File: goodreads.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from .api import search_goodreads
+
+def get_original_date(self, asin: str, authors: str, title: str) -> Dict:
+    api_key = self.config['goodreads_apikey']
+    goodreads_response = search_goodreads(api_key, asin)
+    totalresults = goodreads_get_total_result(goodreads_response)
+
+    if totalresults == 0:
+        #search with author and title
+        #set asin to none to determine that we are querying on author/title
```
Comment (Neurrone) [inline goodreads.py:12]: Where is asin being set to None?

Comment (SenorSmartyPants) [reply goodreads.py:12]: The wording on that comment isn't great. In the first call to `search_goodreads` above, I use ASIN as the only search criteria. When no results are found, I call `search_goodreads` again but without ASIN, just author and title. 

I'll remove that comment.

**Review by Neurrone — APPROVED** [review]

**Pull request merged**